### DISCUSSION TITLE
Asynchronicity?

### DISCUSSION THREAD

**Discussion by Pumbaardian** [n3d1117/chatgpt-telegram-bot#185] [Ideas]
Is it possible to make the code run asynchronously? I know practically nothing about programming, but I can assume that to do this it would be necessary to almost "rewrite" all the existing code? In any case, it would be great if the bot could respond in several dialogs at once, instead of waiting for the last request to complete

**Comment by AlexHTW** [comment]
Hey @Pumbaardian , the bot is asynchronous. It doesn't wait until it's finished with the last request.
What exactly do you mean with "several dialogs"? If you send the bot two messages back to back it will write both answers simultaneously.

**Reply by Pumbaardian** [reply]
Hello! Maybe something is not working correctly for me...
If you write to the bot in two dialogs, it first replies to the first message it receives, and then accepts the second message and replies to it
I will attach some screenshots 
The messages were sent to him at exactly the same time, but he first receives and processes the one that came in a split second faster
![photo_2023-04-06_11-08-37](https://user-images.githubusercontent.com/99145771/230315909-42952b8d-822d-4c98-8bfc-1b6ee59bb4bc.jpg)
![photo_2023-04-06_11-08-46](https://user-images.githubusercontent.com/99145771/230315912-4d50329d-ba55-4cb6-b562-ea8aaf5d665f.jpg)
![photo_2023-04-06_11-08-07](https://user-images.githubusercontent.com/99145771/230315915-2d13f613-9041-44fa-b633-a7edc82d23f8.jpg)
Unfortunately, I am not a native speaker, so I hope I speak clearly enough for you to understand me =)

**Reply by n3d1117** [reply]
Hi @Pumbaardian, I assume you have `STREAM` disabled? With stream enabled, you should start getting both answers simultaneously.

**Reply by Pumbaardian** [reply]
I added this to my .env file
![image](https://user-images.githubusercontent.com/99145771/230377120-73be918c-1b7c-4911-8de6-5a7dfe986425.png)
But nothing has changed. 
The bot still responds to one message first, and only then gets the following 
I'm sorry, this seems to be the issue.

**Reply by Pumbaardian** [reply]
Sorry, I see what the problem is. I'm using bugfloyd's version of the bot with "Enhance Inline Query Functionality with Action Button and Improved Caching".
Apparently, his version, based on the one that didn't have "stream" support yet 
I hope you will do a merge soon with all the new features in the code
Thanks!
Sorry to have bothered you

**Reply by n3d1117** [reply]
All good! Glad you found the issue
